### ISSUE TITLE
Some parsing of xml test output causes build to be marked as failure

### ISSUE THREAD

**Issue by rcurtin** [mlpack/jenkins-conf#8]
It seems like most of the sparc64 builds are failing for this reason.

http://masterblaster.mlpack.org/job/mlpack%20-%20nightly%20matrix%20build/arch=sparc64,armadillo=armadillo-7.500.1,buildmode=debug/1463/console
is an example.

**Comment by zoq** [comment]
What I remember when working on the static analyzer configuration is that the xml parser is very picky about the format (malformed/invalid xml). So I guess:

```
<Exception file="unknown location" line="0">
<![CDATA[
memory access violation at address: 0x00000000: no mapping at fault address
]]>
<LastCheckpoint file="/home/jenkins/workspace/mlpack - nightly matrix 
build/arch/sparc64/armadillo/armadillo-7.500.1/buildmode/debug/repo/src/mlpack/tests/cli_test.cpp" line="107">...</LastCheckpoint>
</Exception>
```

could be the reason. I think if we can reproduce the issue this should be easy to check, I can do that in the next days, unless somebody else likes to work on this.

**Comment by rcurtin** [comment]
Sure, if you want to take a look into it I would appreciate it.  I dug into it a little while back but I could not find exactly what the invalid XML bits were, and I haven't had time to get back to it.  But I think it is fairly low priority; it seems to only happen for the sparc64 builds.

**Comment by zoq** [comment]
It looks like the xUnit plugin can't handle the ``Message`` tag.

error message:

```
Invalid content was found starting with element 'Message'. One of '{TestCase, TestSuite}' is expected.
```

xml part:

```xml
<TestCase name="TestDuplicateFlag" file="/home/jenkins/workspace/mlpack - nightly matrix build/arch/sparc64/armadillo/armadillo-7.950.1/buildmode/release/repo/src/mlpack/tests/cli_test.cpp" line="107">
<Exception file="unknown location" line="0">
    <![CDATA[memory access violation at address: 0x00000000: no mapping at fault address]]>
    <LastCheckpoint file="/home/jenkins/workspace/mlpack - nightly matrix build/arch/sparc64/armadillo/armadillo-7.950.1/buildmode/release/repo/src/mlpack/tests/cli_test.cpp" line="107">
    <![CDATA["TestDuplicateFlag" entry.]]>
</LastCheckpoint>
</Exception>
<Message file="./boost/test/impl/unit_test_log.ipp" line="198">
<![CDATA[Test is aborted]]>
</Message>
<TestingTime>0</TestingTime>
</TestCase>
```
but since the interesting part is the Exception message anyway, I think we can simply remove the node from the output to solve the problem. I've added some lines to the job configuration to test it out.

**Comment by zoq** [comment]
Okay, I guess once we solved the ``TestDuplicateFlag`` issue it should run the other tests as well.

**Comment by rcurtin** [comment]
Nice, thank you for looking into this.  I think it would be a good idea to incorporate this code into `runTestBins.sh`, what do you think?

```
# Remove Message nodes.
cd reports/tests/
wget -O boost_test.py https://gist.githubusercontent.com/zoq/6e61104fae92278b6d4929b6c877ea4b/raw/58b15e3dea7539a23a140750b4eede71d14feccf/boost_test.py
chmod u+x boost_test.py
./boost_test.py > boost_test_tmp.xml
mv boost_test_tmp.xml boost_mlpack_test.xml
cd ../../
```

And we can probably add boost_test.py directly into the repository.  If you think that's a good idea, either I can go ahead and do that or you can, your call.

**Comment by zoq** [comment]
I can integrate the script into ``runTestBins.sh``, unless you can think of a single sed/awk line that can do the same?

**Comment by rcurtin** [comment]
I played with it for a little while, but I had trouble coming up with a single sed line that could capture a single `<Message></Message>` tag properly, and even if I debugged it I'm not sure we wouldn't have problems with it later.  So I think it might be better to just go with the Python script you wrote.

**Comment by zoq** [comment]
Right, I've also tried to come up with something neat but since the ``Message`` tag isn't unique (filename after the tag) it's really ugly. So I ended up with a simple python script.

I'll integrate the script later.

**Comment by zoq** [comment]
Closing, fixed with c800e9f064800265b.
